CMakeLists.txt

```diff
@@ -47,77 +71,79 @@ option (UNITTESTS "Enable unittests generation" OFF)
 
 # Is the code generated for embedded platforms?
 #   1 :   Yes. Matrix update not allowed.
-#   2 :   Yes. Matrix update allowed.
+#   2 :   No. Matrix update allowed.
 if (NOT DEFINED EMBEDDED)
-    message(STATUS "Embedded is OFF")
+    message(STATUS "EMBEDDED = OFF")
 else()
-    message(STATUS "Embedded is ${EMBEDDED}")
+    message(STATUS "EMBEDDED = ${EMBEDDED}")
 endif()
 
-# Is the CUDA support enabled?
-option (CUDA_SUPPORT "Enable CUDA support" OFF)
-message(STATUS "CUDA support is ${CUDA_SUPPORT}")
+option (ALGEBRA_DEFAULT "Default Solver" ON)
+option (ALGEBRA_MKL "Enable MKL Pardiso solver" OFF)
+option (ALGEBRA_CUDA "CUDA Support" OFF)
```

Comment:
Renamed. My thinking was that perhaps going forward we only support `-DALGEBRA=` and use `ALGEBRA_*` booleans only internally (and show a warning if they're specified directly). So for right now I've implemented something like this in the code:
```
set(ALGEBRA "default" CACHE STRING "The Algebra to use (default/mkl/cuda)")

if (DEFINED ALGEBRA_DEFAULT OR DEFINED ALGEBRA_MKL OR DEFINED ALGEBRA_CUDA)
    message(WARNING "Use ALGEBRA variable to specify algebra, not specific ALGEBRA_* boolean variables.")
endif()

if(${ALGEBRA} STREQUAL "default")
    set(ALGEBRA_DEFAULT ON)
    set(ALGEBRA_MKL OFF)
    ...
```
I'll change the usage in CI as well.
Feel free to resolve this conversation if this is acceptable, otherwise we can discuss further.

Reply:
Sounds good, but I would check at the same place if `EMBEDDED` is defined, and if so, force the algebra to default.

Reply:
Done